package.json

```diff
@@ -43,14 +31,13 @@
   ],
   "scripts": {
     "generate-scaffolding": "repo-tools generate all",
-    "prepublishOnly": "node ./prepublish.js",
-    "lint": "eslint ./ test/",
-    "mocha": "mocha -R spec test/*.test.js",
-    "pretest": "node ./prepublish.js",
+    "prepare": "node ./tools/prepublish.js",
+    "prepublish": "npm run prepare",
```

Comment:
I think we want `prepublishOnly` still: https://docs.npmjs.com/misc/scripts#prepublish-and-prepare.

Reply:
Yeah, so this is hella confusing.  What I really want here (I think?) is:
- Run this script after `npm install`
- Run this script before publishing

As of npm 4.x, `prepare` fits that bill.  However, the image in CircleCI for node6 is still using npm 3.x. After we drop node6 support, I am pretty sure I can drop the `prepublish` script. 

Does this sound right?

Reply:
In the past, we just ran pre-publish (unless I'm even hella-ier confused), so that only when we publish this module will it go and fetch the latest proto files. Having that happen on the user's machine at the time of install could be better, but it could have risks, too. Did we already decide that was the best way to go?

Reply:
Technically none of the tests pass unless you run this, so I think we have to do it after npm install 😆

Reply:
Just for tests, we can do a `pre-test` script.

Reply:
Yeah - I mean we could go a lot of ways with this.  I think it was using pre-test, but then we had unit tests that were going out and downloading a big ol tarball every time you run `npm test` :(

Reply:
I'm going to merge for now - but if you feel strongly about this, please let me know!  Happy to change it back if there's something I'm not getting :)

Reply:
> downloading a big ol tarball every time you run npm test :(

With `prepublish`, that will now happen for users when they install this module. The point of our prepublish script was to download the freshest protos from googleapis, then pack it up and include it as part of the package.

The risk of having users do it at the time of install is that each install will get the latest from googleapis@master, which could change irregularly, meaning users will have mismatches irrespective of the `google-proto-files` version.

Reply:
Ahhhhhhh.  I can see why that's bad.  Dumb question: would it make more sense to check the snapshot of the protos into source control?  Then run an npm script every once in a while to pick em back up?